### PR TITLE
Update k8s, prometheus-operator & few other version

### PR DESCRIPTION
**Pull request by Jaanki**
k8s.io/client-go has conflicting dependencies and hence has been
"replaced".

Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

File: pkg/crd/updater.go [hunk 1]
```diff
@@ -96,15 +96,15 @@ func (u *updater) CreateOrUpdateFromEmbedded(ctx context.Context, crdYaml string
 }
 
 func (c *controllerClientCreator) Create(ctx context.Context, crd *apiextensions.CustomResourceDefinition,
-	options metav1.CreateOptions,
+	options metav1.CreateOptions, //nolint:gocritic
```
Comment (tpantelis) [inline pkg/crd/updater.go]:
```suggestion
	options metav1.CreateOptions, // nolint:gocritic // hugeParam - match K8s API
```

File: pkg/crd/updater.go [hunk 2]
```diff
@@ -96,15 +96,15 @@ func (u *updater) CreateOrUpdateFromEmbedded(ctx context.Context, crdYaml string
...
 ) (*apiextensions.CustomResourceDefinition, error) {
 	// TODO skitt handle options
 	err := c.client.Create(ctx, crd)
 	return crd, err
 }
 
 func (c *controllerClientCreator) Update(ctx context.Context, crd *apiextensions.CustomResourceDefinition,
-	options metav1.UpdateOptions,
+	options metav1.UpdateOptions, //nolint:gocritic
```
Comment (tpantelis) [inline pkg/crd/updater.go]:
```suggestion
	options metav1.UpdateOptions, // nolint:gocritic // hugeParam - match K8s API
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Comment by tpantelis** [comment]
@skitt https://github.com/submariner-io/submariner-operator/pull/2180 fixed the system tests! I shouldn't have doubted it 😄 

@Jaanki Still need to fix the nil pointer that's failing the unit tests and address the comments.

**Comment by tpantelis** [comment]
> @Jaanki Still need to fix the nil pointer that's failing the unit tests and address the comments.

OK - I pushed an update to address these.

**Review by tpantelis — APPROVED** [review]

**Comment by tpantelis** [comment]
There's something special going on with OVN. It clones https://github.com/ovn-org/ovn-kubernetes.git but it seems to cause an error with controller-gen related to vendoring:

```
#7 140.7 Error: err: exit status 1: stderr: go: inconsistent vendoring in /go/src/github.com/submariner-io/submariner-operator/ovn-kubernetes/test/e2e/vendor/github.com/spf13/cobra:
#7 140.7 	github.com/cpuguy83/go-md2man/v2@v2.0.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#7 140.7 	github.com/inconshreveable/mousetrap@v1.0.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#7 140.7 	github.com/spf13/pflag@v1.0.5: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#7 140.7 	gopkg.in/yaml.v2@v2.4.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
#7 140.7 
#7 140.7 	To ignore the vendor directory, use -mod=readonly or -mod=mod.
#7 140.7 	To sync the vendor directory, run:
#7 140.7 		go mod vendor
```

even though `go mod vendor` was run earlier.

**Comment by skitt** [comment]
> @skitt #2180 fixed the system tests! I shouldn't have doubted it :smile:

I wasn’t entirely sure it would either — your comment made sense when looking at the code change! I’m pretty sure I’d found a wonky code path that my PR fixed, where controllers were started too soon, but I couldn’t find it.

**Comment by skitt** [comment]
> There's something special going on with OVN. It clones https://github.com/ovn-org/ovn-kubernetes.git but it seems to cause an error with controller-gen related to vendoring:
> 
> ```
> #7 140.7 Error: err: exit status 1: stderr: go: inconsistent vendoring in /go/src/github.com/submariner-io/submariner-operator/ovn-kubernetes/test/e2e/vendor/github.com/spf13/cobra:
> #7 140.7 	github.com/cpuguy83/go-md2man/v2@v2.0.1: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
> #7 140.7 	github.com/inconshreveable/mousetrap@v1.0.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
> #7 140.7 	github.com/spf13/pflag@v1.0.5: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
> #7 140.7 	gopkg.in/yaml.v2@v2.4.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
> #7 140.7 
> #7 140.7 	To ignore the vendor directory, use -mod=readonly or -mod=mod.
> #7 140.7 	To sync the vendor directory, run:
> #7 140.7 		go mod vendor
> ```
> 
> even though `go mod vendor` was run earlier.

Ah, that’s probably the generic vendoring I added to deal with multiple modules — we’re probably trying to run `go mod vendor` in modules that we should leave untouched.

**Comment by tpantelis** [comment]
> Ah, that’s probably the generic vendoring I added to deal with multiple modules — we’re probably trying to run `go mod vendor` in modules that we should leave untouched.

Yup.  I was able to fix it by filtering out "ovn-kubernetes" from `VENDOR_MODULES`.

**Comment by tpantelis** [comment]
> Yup. I was able to fix it by filtering out "ovn-kubernetes" from VENDOR_MODULES.

I lied. It still fails with "inconsistent vendoring" in _ovn-kubernetes/test/e2e_ even though we didn't run `go mod vendor`. Somehow it seems there's still a vendor directory generated (it's not checked into [source](https://github.com/ovn-org/ovn-kubernetes/tree/master/test/e2e)). I'll try hacking it to remove the vendor directory.

**Comment by tpantelis** [comment]
I verified via `ls` prior to running `controller-gen` that the vendor dir doesn't exist:

`ls: cannot access '/go/src/github.com/submariner-io/submariner-operator/ovn-kubernetes/test/e2e/vendor': No such file or directory`

So I'm at a loss as to why it still complains about "inconsistent vendoring". Locally, if the vendor dir doesn't exist, `controller-gen` runs fine.

File: controllers/helpers/helpers.go [hunk 3]
```diff
@@ -268,3 +268,13 @@ func IsImmutableError(err error) bool {
 
 	return false
 }
+
+func setLabels(from, to controllerClient.Object) {
```
Comment (mkolesnik) [inline controllers/helpers/helpers.go]:
```suggestion
func copyLabels(from, to controllerClient.Object) {
```

Makes more sense and easier to understand

Comment (tpantelis) [reply controllers/helpers/helpers.go]: Done

**Review by mkolesnik — APPROVED** [review]
LGTM

**Review by skitt — APPROVED** [review]

**Pull request merged**